include/pm/device_runtime.h

```diff
@@ -8,107 +9,104 @@
+ * @defgroup runtime_power_management_api Device Runtime Power Management API
  * @ingroup power_management_api
  * @{
  */
 
-#ifdef CONFIG_PM_DEVICE_RUNTIME
-
+#if defined(CONFIG_PM_DEVICE_RUNTIME) || defined(__DOXYGEN__)
 /**
  * @brief Enable device runtime PM
  *
- * Called by a device driver to enable device runtime power management.
- * The device might be asynchronously suspended if runtime PM is enabled
- * when the device is not use.
- *
  * @funcprops \pre_kernel_ok
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
+ * @param dev Device instance.
  */
 void pm_device_enable(const struct device *dev);
 
 /**
  * @brief Disable device runtime PM
  *
- * Called by a device driver to disable device runtime power management.
- * The device might be asynchronously resumed if runtime PM is disabled
+ * If the device is currently suspended it will be resumed.
  *
  * @funcprops \pre_kernel_ok
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
+ * @param dev Device instance.
+ *
+ * @retval 0 If the device runtime PM is disabled successfully.
+ * @retval -ENOSYS If the functionality is not available.
+ * @retval -errno Other negative errno, result of resuming the device.
  */
-void pm_device_disable(const struct device *dev);
+int pm_device_disable(const struct device *dev);
 
 /**
- * @brief Call device resume synchronously based on usage count
+ * @brief Resume a device based on usage count.
  *
- * Called by a device driver to mark the device as being used. It
- * will bring up or resume the device if it is in suspended state
- * based on the device usage count. This call is blocked until the
- * device PM state is changed to resume.
+ * This function will increase device usage count. If the device is suspended,
+ * it will be resumed. In case of failure, usage count and device state will be
+ * left unchanged.
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
- * @retval 0 If successful.
- * @retval Errno Negative errno code if failure.
+ * @funcprops \pre_kernel_ok
+ *
+ * @param dev Device instance.
+ *
+ * @retval 0 If the device has been resumed successfully.
+ * @retval -ENOSTUP If runtime PM is not enabled for the device.
+ * @retval -ENOSYS If the functionality is not available.
+ * @retval -errno Other negative errno, result of the PM action callback.
  */
 int pm_device_get(const struct device *dev);
 
 /**
- * @brief Call device suspend asynchronously based on usage count
+ * @brief Suspend a device based on usage count.
  *
- * Called by a device driver to mark the device as being released.
- * This API asynchronously put the device to suspend state if
- * it not already in suspended state.
+ * This function will decrease usage count. If usage count reaches zero, the
+ * device will be suspended. In case of failure, usage count and device state
```

Comment:
Ditto: "Suspend the device, based on usage count. Device usage count is decreased. Suspension happens if counter reaches zero"